org/diceresearch/nebula/orchestrator.py

```diff
@@ -2,140 +2,141 @@
 
-import claimworthinesschecker
-import claimworthinesscheckerdummy
-import databasemanager
-import evidenceretrieval
+from claim_worthiness_check import dummy_claim_check, claim_buster
+from coref_resolution import coreference_resolution
+from database import databasemanager
+from evidence_retrieval import elastic_search
 import settings
-import stancedetection
-import translator
+from stance_detection import cosine_similarity
+from translation import neamt_translator
+from indicators.main import run_indicator_check_text
+from utils.util import SetEncoder
 from veracity_detection import predictions
 
 from utils import notification
 
 
 def goNextLevel(identifier):
-    logging.info("Orch:" + identifier)
+    """
+    Increments and executes next stage
+    :param identifier: ID
+    :return:
+    """
     current = databasemanager.getOne(settings.results_table_name, identifier)
-    STAGE_NUMBER = current[1]
-    INPUT_TEXT = current[2]
-    INPUT_LANG = current[3]
-    TRANSLATED_TEXT = current[4]
-    CLAIM_CHECK_WORTHINESS_RESULT = current[6]
-    EVIDENCE_RETRIEVAL_RESULT = current[8]
-    STANCE_DETECTION_RESULT = current[10]
-
-    #notification
-    REGISTRATION_TOKEN = current[14]
+
+    stage_number = current[1]
+    input_lang = current[2]
+    input_text = current[3]
+    translated_text = current[4]
+    coref_text = current[5]
+    sentences = current[14]
+    veracity_label = current[20]
+
+    # If the record could not be found
+    if current is None:
+        logging.error("Could not find this row in database {}".format(identifier))
+
+    if sentences:
+        claims = json.loads(sentences)
+    # Increase fact checking step
+    current_stage = int(stage_number)
+
+    # notification
+    REGISTRATION_TOKEN = current[16]
     NOTIFICATION_TITLE = "Your result is ready!"
-    NOTIFICATION_BODY = f"Your result with ID {identifier} is now available.",
+    NOTIFICATION_BODY = "Your result with ID {} is now available.".format(identifier),
 
-    if current == None:
-        logging.info("Could not find this row in database " + identifier)
-    current_stage = int(STAGE_NUMBER)
-    logging.info("Current stage is : " + str(current_stage))
     next_stage = current_stage + 1
+    logging.debug("Current stage is : {}".format(current_stage))
+
     if next_stage == 1:
+        logging.info("Orch: {}".format(identifier))
+        logging.debug("Translation")
+
+        # set status to ongoing and set timestamp
         databasemanager.update_step(settings.results_table_name, settings.status, settings.ongoing, identifier)
         databasemanager.update_step(settings.results_table_name, settings.timestamp, datetime.now().isoformat('#'),
                                     identifier)
-        # translate
-        # language
-        if INPUT_LANG != "en":
+
+        # translate if language differs from english and we were not provided a translation already
+        if input_lang != "en" and not translated_text:
```

Comment:
I think this collides with the previously defined default value "nd". The translation step may try to translate from not defined to English, or? :thinking:

Reply:
How does it collide? 

> The translation step may try to translate from not defined to English, or?

Shouldn't that be the case? We need to have the text in english for the next steps.